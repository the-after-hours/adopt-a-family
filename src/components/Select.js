import React from 'react';
import PropTypes from 'prop-types';

import options from '../constants/options';

const Select = props => {
  const dropdownItems = options[props.menuType].map(optionItem => {
    return (
      <option key={optionItem.toLowerCase()} value={optionItem.toLowerCase()}>
        {optionItem}
      </option>
    );
  });

  return (
    <select form={props.form} name={props.menuType}>
      {dropdownItems}
    </select>
  );
};

Select.propTypes = {
  form: PropTypes.string,
  menuType: PropTypes.string,
  optionItem: PropTypes.string
};

export default Select;