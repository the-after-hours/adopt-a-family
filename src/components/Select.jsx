import React from 'react';

const Select = (props) => (
  <select name={props.name}>
    <option value="donator">Donator</option>
    <option value="family">Family</option>
    <option value="organizer">Organizer</option>
  </select>
);

export default Select;
