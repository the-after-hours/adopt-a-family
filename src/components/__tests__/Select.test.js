import faker from 'faker';
import React from 'react';
import renderer from 'react-test-renderer';

import options from '../../constants/options';
import Select from '../Select';

describe('Select', () => {
  test('Select snapshot', () => {
    const component = renderer.create(<Select menuType="states"/>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('Drop down options', () => {
    it('Should pass in an array', () => {
      const objectKeys = Object.keys(options);

      objectKeys.map((key) => {
        expect(Array.isArray(options[key])).toBeTruthy();
      });
    });
  });

  describe('States', () => {
    it('Should contain valid states', () => {
      const stateAbbr = faker.address.stateAbbr();

      expect(options.states.indexOf(stateAbbr)).not.toBe(0);
    });
  });
});
