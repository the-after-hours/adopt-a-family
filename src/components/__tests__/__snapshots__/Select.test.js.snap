// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Select snapshot 1`] = `
<select
  form={undefined}
  name="states"
>
  <option
    value="ak"
  >
    AK
  </option>
  <option
    value="al"
  >
    AL
  </option>
  <option
    value="az"
  >
    AZ
  </option>
  <option
    value="ar"
  >
    AR
  </option>
  <option
    value="ca"
  >
    CA
  </option>
  <option
    value="co"
  >
    CO
  </option>
  <option
    value="ct"
  >
    CT
  </option>
  <option
    value="de"
  >
    DE
  </option>
  <option
    value="fl"
  >
    FL
  </option>
  <option
    value="ga"
  >
    GA
  </option>
  <option
    value="hi"
  >
    HI
  </option>
  <option
    value="id"
  >
    ID
  </option>
  <option
    value="il"
  >
    IL
  </option>
  <option
    value="in"
  >
    IN
  </option>
  <option
    value="ia"
  >
    IA
  </option>
  <option
    value="ks"
  >
    KS
  </option>
  <option
    value="ky"
  >
    KY
  </option>
  <option
    value="la"
  >
    LA
  </option>
  <option
    value="me"
  >
    ME
  </option>
  <option
    value="md"
  >
    MD
  </option>
  <option
    value="ma"
  >
    MA
  </option>
  <option
    value="mi"
  >
    MI
  </option>
  <option
    value="mn"
  >
    MN
  </option>
  <option
    value="ms"
  >
    MS
  </option>
  <option
    value="mo"
  >
    MO
  </option>
  <option
    value="mt"
  >
    MT
  </option>
  <option
    value="ne"
  >
    NE
  </option>
  <option
    value="nv"
  >
    NV
  </option>
  <option
    value="nh"
  >
    NH
  </option>
  <option
    value="nj"
  >
    NJ
  </option>
  <option
    value="nm"
  >
    NM
  </option>
  <option
    value="ny"
  >
    NY
  </option>
  <option
    value="nc"
  >
    NC
  </option>
  <option
    value="nd"
  >
    ND
  </option>
  <option
    value="oh"
  >
    OH
  </option>
  <option
    value="ok"
  >
    OK
  </option>
  <option
    value="or"
  >
    OR
  </option>
  <option
    value="pa"
  >
    PA
  </option>
  <option
    value="ri"
  >
    RI
  </option>
  <option
    value="sc"
  >
    SC
  </option>
  <option
    value="sd"
  >
    SD
  </option>
  <option
    value="tn"
  >
    TN
  </option>
  <option
    value="tx"
  >
    TX
  </option>
  <option
    value="ut"
  >
    UT
  </option>
  <option
    value="vt"
  >
    VT
  </option>
  <option
    value="va"
  >
    VA
  </option>
  <option
    value="wa"
  >
    WA
  </option>
  <option
    value="wv"
  >
    WV
  </option>
  <option
    value="wi"
  >
    WI
  </option>
  <option
    value="wy"
  >
    WY
  </option>
</select>
`;
